[Bug]: Feishu plugin still prompts to download after installation

### Bug type

Behavior bug (incorrect output/state without crash)

### Summary

### English

When adding a Feishu channel, the system prompts to download the Feishu plugin (@openclaw/feishu) for the first time. However, after downloading, the system still prompts that the plugin needs to be downloaded, even though the download has been completed.
After that, it keeps prompting to download repeatedly.

### Simplified Chinese

在添加飞书频道时，第一次提示下载飞书插件，然而下载之后系统仍然提示需要下载插件，尽管插件已经下载完成。
然后，一直这样循环。

### Steps to reproduce

run: 
```bash
openclaw onboard
```

Skip operations such as configuration models.

1. Download and install the OpenClaw Feishu plugin (@openclaw/feishu)
```text
◆  Install Feishu plugin?
│  ● Download from npm (@openclaw/feishu)
│  ○ Skip for now
└
```
2. Select a channel: ● Feishu/Lark (飞书) , Still showing download options：
```text
◆  Install Feishu plugin?
│  ● Download from npm (@openclaw/feishu)
│  ○ Skip for now
└
```

### Expected behavior

After downloading the plugin, it should work properly without repeatedly prompting to download.

### Actual behavior

Even though the plugin has been downloaded, the installation prompt still appears, preventing the addition of Feishu channel.

### OpenClaw version

2026.3.7

### Operating system

Debian GNU/Linux 13 (trixie)

### Install method

npm install -g openclaw@latest

### Logs, screenshots, and evidence

```shell
◇  Select a channel
│  Feishu/Lark (飞书)
│
◇  Install Feishu plugin?
│  Download from npm (@openclaw/feishu)
Downloading @openclaw/feishu…
Extracting /tmp/openclaw-npm-pack-cukQFy/openclaw-feishu-2026.3.7.tgz…
WARNING: Plugin "feishu" contains dangerous code patterns: Environment variable access combined with network send — possible credential harvesting (/tmp/openclaw-plugin-kM4mJv/extract/package/src/client.test.ts:74); Environment variable access combined with network send — possible credential harvesting (/tmp/openclaw-plugin-kM4mJv/extract/package/src/client.ts:12)
Installing to /home/enlin/.openclaw/extensions/feishu…
Installing plugin dependencies…
16:38:33 [plugins] plugins.allow is empty; discovered non-bundled plugins may auto-load: feishu (/home/enlin/.openclaw/extensions/feishu/index.ts). Set plugins.allow to explicit trusted ids.
16:38:34 [plugins] feishu failed to load from /home/enlin/.nvm/versions/node/v22.16.0/lib/node_modules/openclaw/extensions/feishu/index.ts: Error: Cannot find module '@larksuiteoapi/node-sdk'
Require stack:
- /home/enlin/.nvm/versions/node/v22.16.0/lib/node_modules/openclaw/extensions/feishu/src/client.ts
│
◇  Channel setup ───────────────────────────╮
│                                           │
│  feishu does not support onboarding yet.  │
│                                           │
├───────────────────────────────────────────╯
│
◇  Select a channel
│  Feishu/Lark (飞书)
│
◇  Install Feishu plugin?
│  Skip for now
│
◇  Select a channel
│  Finished
Config warnings:
- plugins.entries.feishu: plugin feishu: duplicate plugin id detected; later plugin may be overridden (/home/enlin/.openclaw/extensions/feishu/index.ts)
Config overwrite: /home/enlin/.openclaw/openclaw.json (sha256 e3da3b21baa1fe92dcad3dc810db9136826701d49e95236e69ede95caa8918a1 -> 063b211116a3b6f42453e68a1e33f0eed91b61ec2cc1a663512e76cc5bc9a38f, backup=/home/enlin/.openclaw/openclaw.json.bak)
Updated ~/.openclaw/openclaw.json
Config warnings:\n- plugins.entries.feishu: plugin feishu: duplicate plugin id detected; later plugin may be overridden (/home/enlin/.openclaw/extensions/feishu/index.ts)
│
◇  Control UI ────────────────────────────────────╮
│                                                 │
│  Web UI: http://127.0.0.1:18789/                │
│  Gateway WS: ws://127.0.0.1:18789               │
│  Gateway: reachable                             │
│  Docs: https://docs.openclaw.ai/web/control-ui  │
│                                                 │
├─────────────────────────────────────────────────╯
│
└  Configure complete.
```

### Impact and severity

_No response_

### Additional information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,7 @@ Docs: https://docs.openclaw.ai
 
 - Docker/runtime image: prune dev dependencies, strip build-only dist metadata for smaller Docker images. (#40307) Thanks @vincentkoc.
 - Plugins/channel onboarding: prefer bundled channel plugins over duplicate npm-installed copies during onboarding and release-channel sync, preventing bundled plugins from being shadowed by npm installs with the same plugin ID. (#40092)
+- Feishu/plugin onboarding: clear the short-lived plugin discovery cache before reloading the registry after installing a channel plugin, so onboarding no longer re-prompts to download Feishu immediately after a successful install. Fixes #39642. (#39752) Thanks @GazeKingNuWu.
 - macOS app/chat UI: route browser proxy through the local node browser service, preserve plain-text paste semantics, strip completed assistant trace/debug wrapper noise from transcripts, refresh permission state after returning from System Settings, and tolerate malformed cron rows in the macOS tab. (#39516) Thanks @Imhermes1.
 - Mattermost replies: keep `root_id` pinned to the existing thread root when an agent replies inside a thread, while still using reply-target threading for top-level posts. (#27744) thanks @hnykda.
 - Agents/failover: detect Amazon Bedrock `Too many tokens per day` quota errors as rate limits across fallback, cron retry, and memory embeddings while keeping context-window `too many tokens per request` errors out of the rate-limit lane. (#39377) Thanks @gambletan.

--- a/src/commands/onboarding/plugin-install.test.ts
+++ b/src/commands/onboarding/plugin-install.test.ts
@@ -49,12 +49,21 @@ vi.mock("../../plugins/loader.js", () => ({
   loadOpenClawPlugins: vi.fn(),
 }));
 
+const clearPluginDiscoveryCache = vi.fn();
+vi.mock("../../plugins/discovery.js", () => ({
+  clearPluginDiscoveryCache: () => clearPluginDiscoveryCache(),
+}));
+
 import fs from "node:fs";
 import type { ChannelPluginCatalogEntry } from "../../channels/plugins/catalog.js";
 import type { OpenClawConfig } from "../../config/config.js";
+import { loadOpenClawPlugins } from "../../plugins/loader.js";
 import type { WizardPrompter } from "../../wizard/prompts.js";
 import { makePrompter, makeRuntime } from "./__tests__/test-utils.js";
-import { ensureOnboardingPluginInstalled } from "./plugin-install.js";
+import {
+  ensureOnboardingPluginInstalled,
+  reloadOnboardingPluginRegistry,
+} from "./plugin-install.js";
 
 const baseEntry: ChannelPluginCatalogEntry = {
   id: "zalo",
@@ -235,5 +244,28 @@ describe("ensureOnboardingPluginInstalled", () => {
     expectPluginLoadedFromLocalPath(result);
     expect(note).toHaveBeenCalled();
     expect(runtime.error).not.toHaveBeenCalled();
+  });
+
+  it("clears discovery cache before reloading the onboarding plugin registry", () => {
+    const runtime = makeRuntime();
+    const cfg: OpenClawConfig = {};
+
+    reloadOnboardingPluginRegistry({
+      cfg,
+      runtime,
+      workspaceDir: "/tmp/openclaw-workspace",
+    });
+
+    expect(clearPluginDiscoveryCache).toHaveBeenCalledTimes(1);
+    expect(loadOpenClawPlugins).toHaveBeenCalledWith(
+      expect.objectContaining({
+        config: cfg,
+        workspaceDir: "/tmp/openclaw-workspace",
+        cache: false,
+      }),
+    );
+    expect(clearPluginDiscoveryCache.mock.invocationCallOrder[0]).toBeLessThan(
+      vi.mocked(loadOpenClawPlugins).mock.invocationCallOrder[0] ?? Number.POSITIVE_INFINITY,
+    );
   });
 });

--- a/src/commands/onboarding/plugin-install.ts
+++ b/src/commands/onboarding/plugin-install.ts
@@ -4,12 +4,12 @@ import { resolveAgentWorkspaceDir, resolveDefaultAgentId } from "../../agents/ag
 import type { ChannelPluginCatalogEntry } from "../../channels/plugins/catalog.js";
 import { resolveBundledInstallPlanForCatalogEntry } from "../../cli/plugin-install-plan.js";
 import type { OpenClawConfig } from "../../config/config.js";
-import { clearPluginDiscoveryCache } from "../../plugins/discovery.js";
 import { createSubsystemLogger } from "../../logging/subsystem.js";
 import {
   findBundledPluginSourceInMap,
   resolveBundledPluginSources,
 } from "../../plugins/bundled-sources.js";
+import { clearPluginDiscoveryCache } from "../../plugins/discovery.js";
 import { enablePluginInConfig } from "../../plugins/enable.js";
 import { installPluginFromNpmSpec } from "../../plugins/install.js";
 import { buildNpmResolutionInstallFields, recordPluginInstall } from "../../plugins/installs.js";

--- a/src/commands/onboarding/plugin-install.ts
+++ b/src/commands/onboarding/plugin-install.ts
@@ -4,6 +4,7 @@ import { resolveAgentWorkspaceDir, resolveDefaultAgentId } from "../../agents/ag
 import type { ChannelPluginCatalogEntry } from "../../channels/plugins/catalog.js";
 import { resolveBundledInstallPlanForCatalogEntry } from "../../cli/plugin-install-plan.js";
 import type { OpenClawConfig } from "../../config/config.js";
+import { clearPluginDiscoveryCache } from "../../plugins/discovery.js";
 import { createSubsystemLogger } from "../../logging/subsystem.js";
 import {
   findBundledPluginSourceInMap,
@@ -224,6 +225,7 @@ export function reloadOnboardingPluginRegistry(params: {
   runtime: RuntimeEnv;
   workspaceDir?: string;
 }): void {
+  clearPluginDiscoveryCache();
   const workspaceDir =
     params.workspaceDir ?? resolveAgentWorkspaceDir(params.cfg, resolveDefaultAgentId(params.cfg));
   const log = createSubsystemLogger("plugins");

--- a/src/cron/isolated-agent/run.test-harness.ts
+++ b/src/cron/isolated-agent/run.test-harness.ts
@@ -64,6 +64,7 @@ vi.mock("../../agents/skills/refresh.js", () => ({
 }));
 
 vi.mock("../../agents/workspace.js", () => ({
+  DEFAULT_IDENTITY_FILENAME: "IDENTITY.md",
   ensureAgentWorkspace: vi.fn().mockResolvedValue({ dir: "/tmp/workspace" }),
 }));
 

